components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.ui/src/main/java/org/wso2/carbon/event/output/adapter/ui/UIEventAdapter.java

```diff
@@ -104,14 +121,22 @@ public void init() throws OutputEventAdapterException {
             }
 
             executorService = new ThreadPoolExecutor(minThread, maxThread, defaultKeepAliveTime, TimeUnit.MILLISECONDS,
-                    new LinkedBlockingQueue<Runnable>(jobQueSize));
+                                                     new LinkedBlockingQueue<Runnable>(jobQueSize));
         }
 
         streamId = eventAdapterConfiguration.getOutputStreamIdOfWso2eventMessageFormat();
         if (streamId == null || streamId.isEmpty()) {
             throw new OutputEventAdapterRuntimeException("UI event adapter needs a output stream id");
         }
 
+        // fetch the "streamDefinition" corresponding to the "streamId" and then fetch the different attribute types
+        // of the streamDefinition corresponding to the event's streamId. They are required when validating values in
+        // the events against the streamDef attributes.
+        streamDefinition = getStreamDefinition(streamId);
+        streamMetaAttributes = streamDefinition.getMetaData();
+        streamCorrelationAttributes = streamDefinition.getCorrelationData();
+        streamPayloadAttributes = streamDefinition.getPayloadData();
```

Comment:
@Shabirmean It seems you are setting the stream attributes at the time the adapter is created. If the stream definition is changed, will the changes be propagated to the adapter fields? Were you able to test this implementation by changing the stream definition as well?